teuthology/task/install/rpm.py

```diff
@@ -194,28 +198,26 @@ def _update_package_list_and_install(ctx, remote, rpm, config):
                 cpack=cpack,
             )
             remote.run(
-                args=['if', 'test', '-e',
+                args=list(_flat2gen(['if', 'test', '-e',
                       run.Raw(pkg), run.Raw(';'), 'then',
                       'sudo', pkg_mng_cmd, pkg_mng_opts, 'remove',
-                      pkg_mng_subcommand_opts, pkg, run.Raw(';'),
-                      'sudo', pkg_mng_cmd, pkg_mng_opts, pkg_mng_gpg_opt, 'install',
-                      pkg_mng_subcommand_opts, pkg_mng_install_opts,
-                      pkg, run.Raw(';'),
-                      'fi']
-            )
+                      pkg_mng_remove_opts, pkg, run.Raw(';'),
+                      'sudo', pkg_mng_cmd, pkg_mng_opts, 'install',
+                      pkg_mng_install_opts, pkg, run.Raw(';'),
+                      'fi'])))
         if pkg is None:
-            remote.run(args=[
-                'sudo', pkg_mng_cmd, pkg_mng_opts, pkg_mng_gpg_opt, 'install',
-                pkg_mng_subcommand_opts, pkg_mng_install_opts, cpack
-            ])
+            remote.run(args=list(_flat2gen([
+                'sudo', pkg_mng_cmd, pkg_mng_opts, 'install',
+                pkg_mng_install_opts, cpack
+            ])))
         else:
             remote.run(
-                args=['if', 'test', run.Raw('!'), '-e',
+                args=list(_flat2gen(['if', 'test', run.Raw('!'), '-e',
```

Comment:
Do you understand why here is another try to install package?

Reply:
And there is no try to remove it.

Reply:
Yes, I understand (more or less), but this `else` block is never exercised by us. It's only used when `config` contains a key called `local`. So it has no bearing on the bug this PR is fixing.